nuclear_simulator/data_gen/runners/maintenance_scenario_runner.py

```
@@ -0,0 +1,1498 @@
+            
+            if self.verbose:
+                print(f"   ✅ Successfully loaded YAML configuration")
+                print(f"   📊 Configuration sections: {len(loaded_config)}")
+            
+            # Validate the loaded configuration
+            self._validate_config(loaded_config)
+            
+            return loaded_config
+            
+        except yaml.YAMLError as e:
+            raise yaml.YAMLError(f"Error parsing YAML file {yaml_path}: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)